### ISSUE TITLE
Standard Config Breaks object-property-newline

### ISSUE THREAD

**Issue by srm985** [eslint/eslint#11691]
**Tell us about your environment**

* **ESLint Version:** 5.16.0
* **Node Version:** 10.15.3
* **npm Version:** 6.5.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
// .eslintrc.json

{
    "extends": "standard",
    "rules": {
        "object-property-newline": 2
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

As a proof of concept, I simply ran `eslint --init`. I then selected the `standard` configuration to extend. From there I added the rule` "object-property-newline": 2` in the `.eslintrc.json` file. I then added a npm script command of `lint: "eslint test.js --fix"` and ran `npm run lint`. If I remove the extended standard config from the `.eslintrc.json` file, the `object-property-newline` works as expected.

```js
// package.json

{
  "name": "object-property-newline",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "lint": "eslint test.js --fix"
  },
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "eslint": "^5.16.0",
    "eslint-config-standard": "^12.0.0",
    "eslint-plugin-import": "^2.17.2",
    "eslint-plugin-node": "^9.0.1",
    "eslint-plugin-promise": "^4.1.1",
    "eslint-plugin-standard": "^4.0.0"
  }
}
```

```js
// test.js

var a = { b: 2, c: 3 }

console.log(a)
```

```bash
eslint test.js --fix
```

**What did you expect to happen?**
As the fix flag was set to `true`, I expected the file to be modified as such:

```js
// test.js

var a = { b: 2, 
c: 3 }

console.log(a)
```

**What actually happened? Please include the actual, raw output from ESLint.**
If `"extends": "standard"` is used, the rule `object-property-newline` does not trigger an error nor does it fix the layout. If `"extends": "standard"` is removed, the rule works as expected.

```js
// test.js

var a = { b: 2, c: 3 }

console.log(a)
```

**Are you willing to submit a pull request to fix this bug?**
Possibly

**Comment by aladdin-add** [comment]
emm.. interesting! seems like a bug to me.

**Comment by aladdin-add** [comment]
the reason was eslint merged your severity and the rule options from `eslint-config-standard`.
so it is expected, according to these lines: https://github.com/eslint/eslint/blob/d662b178c7dad193201564d16f7977af2f81ebcf/lib/cli-engine/config-array/config-array.js#L213-L216

**Comment by kaicataldo** [comment]
@srm985 To turn it off immediately, you should be able to add
```
"object-property-newline": ["error", { "allowMultiplePropertiesPerLine": false }],
```
to your configuration, since the options are merged (and Standard sets `"allowMultiplePropertiesPerLine": true`).

I do think that this behavior is unintuitive (even if it is intended). It makes sense that we merge these configs, because it would be rather tedious for to have to redefine all envs, plugins, etc. you'd like to share with the extended config. We could potentially just not merge rules, but then the merging behavior is even more complex and potentially confusing.

**Comment by srm985** [comment]
@aladdin-add thanks for investigating. 

@kaicataldo thanks for the workaround. That makes sense.
